### DISCUSSION TITLE
How to pass incoming message to global bot.hears handlers from a scene?

### DISCUSSION THREAD

**Discussion by centuriononon** [telegraf/telegraf#2049] [Q&A] [answered]
I have a menu with commands that should be processed by global `bot.hears` handlers. When entering a scene, the menu buttons remain visible, but clicking a button should trigger the corresponding global handler **outside the scene**.

In the last step of my WizardScene, I use a fallback function that exits the scene and is supposed to **pass the command to the global handlers for processing**. _However, the commands are still handled within the scene instead of being passed upward._

**How can I make this work as intended?**

**Comment by MKRhere** [comment, marked as answer]
What you could do is to add bypass handlers to stage instead of bot.

```TS
const bot = new Telegraf(token);

bot.use(session());

const stage = new Scenes.Stage([Scene1, Wizard1, ...]);

stage.hears(/^Exit$/, ctx => {
	// perhaps other logic
	return ctx.scene.leave();
});

bot.use(stage);
```

The advantages of attaching to stage are twofold:
1. These handlers run AFTER `ctx.scene` is set
2. But BEFORE any scene is processed
